haddock-library/src/Documentation/Haddock/Parser/Util.hs

```diff
@@ -13,60 +13,73 @@
 module Documentation.Haddock.Parser.Util (
   unsnoc
 , strip
+, inClass
+, notInClass
 , takeUntil
 , removeEscapes
 , makeLabeled
 , takeHorizontalSpace
 , skipHorizontalSpace
 ) where
 
+import qualified Data.Text as T
+import           Data.Text (Text)
+
 import           Control.Applicative
 import           Control.Monad (mfilter)
-import           Documentation.Haddock.Parser.Monad hiding (isHorizontalSpace)
-import           Data.ByteString.Char8 (ByteString)
-import qualified Data.ByteString.Char8 as BS
+import           Documentation.Haddock.Parser.Monad
 import           Prelude hiding (takeWhile)
 
-#if MIN_VERSION_bytestring(0,10,2)
-import           Data.ByteString.Char8 (unsnoc)
-#else
-unsnoc :: ByteString -> Maybe (ByteString, Char)
+import           Data.Maybe (isJust)
+import           Data.Char (isSpace)
+
+-- | Like 'T.uncons', but for the last character instead of the first.
```

Comment:
Precisely this function already exists in `text` and is omgoptimized https://hackage.haskell.org/package/text-1.2.3.0/docs/Data-Text.html#v:unsnoc

Reply:
I swear I looked for this, but didn't find it. *fpalm